var/spack/repos/builtin/packages/libf77zmq/package.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+from spack import *
+
+
+class Libf77zmq(MakefilePackage):
+    """Fortran binding for the ZeroMQ communication library"""
+
+    homepage = "http://zguide.zeromq.org/"
+    url      = "https://github.com/zeromq/f77_zmq/archive/4.3.1.tar.gz"
+
+    maintainers = ['scemama']
+
+    version('4.3.1', sha256='a15d72d93022d3e095528d2808c7767cece974a2dc0e2dd95e4c122f60fcf0a8')
+
+    depends_on('libzmq')
+    depends_on('python', type='build')
+
+    def setup_build_environment(self, env):
+        env.append_flags('CFLAGS', '-O3')
+        env.append_flags('CFLAGS', '-g')
+
+    def edit(self, spec, prefix):
+        cflags = os.environ.get('CFLAGS')
+        makefile = FileFilter('Makefile')
+        makefile.filter('CC=.*',
+                        'CC={0} {1}'.format(self.compiler.cc,
```

Comment:
This will use the actual compiler instead of Spack's compiler wrapper. You should use:
```suggestion
                        'CC={0} {1}'.format(spack_cc,
```

Reply:
I saw many packages using `self.compiler.cc` and I don't catch the difference between `spack_cc` and `self.compiler.cc`. Can you point me to some documentation explaining the difference between those two?

Reply:
I'm not sure how well documented it is...

`self.compiler.cc` is the actual compiler you register in `compilers.yaml`. `spack_cc` is Spack's compiler wrapper (`lib/spack/env/cc`), which wraps around `self.compiler.cc`. The compiler wrapper is responsible for things like automatically injecting `-L/path/to/lib`, `-Wl,-rpath`, and `-I/path/to/include` for all dependencies. Any package that uses `self.compiler.cc` is either:

1. Broken, it should be using `spack_cc`
2. Not used during build time, it replaces `spack_cc` with `self.compiler.cc` after installation
3. Uses SCons or some other awful build system that ignores environment variables, making our compiler wrappers useless

Reply:
OK, thanks a lot!